tests/lib_test.php

```diff
@@ -62,6 +62,11 @@ public function get_data_for_pattern_matching() {
             array('https://my_moodle.com/', 'https://my_moodle.com//', false),
             array('https://my_moodle.com/', 'http://my_moodle.com', false),
             array('https://my_moodle.com/', 'http://my_moodle.com', false),
+            array('https://my_moodle.com/', '/', true),
+            array('https://my_moodle.com/', 'o{2}', true),
+            array('https://my_moodle.com/', 'o{3}', false),
+            array('https://my_moodle3.com/', 'https://my_moodle[1-5].com', true),
+            array('https://my_moodle6.com/', 'https://my_moodle[1-5].com', false),
```

Comment:
can you also add some tests for other common regex things like parenthesis ( ) and in particular ?, I'm trying to think of other valid http chars which might mess up a regex?